build/common.sh

```diff
@@ -628,11 +632,15 @@ function kube::build::rsync_probe {
 # This will set the global var KUBE_RSYNC_ADDR to the effective port that the
 # rsync daemon can be reached out.
 function kube::build::start_rsyncd_container() {
+  IPTOOL=ifconfig
+  if kube::build::has_ip ; then
```

Comment:
This prefers ip over ifconfig. Also, which "ip" tool are you referring to? The name is pretty generic. Does it accept the same syntax? I would want to try ifconfig first, and ifconfig doesn't work, then try another tool.

Reply:
`ifconfig` comes from the `net-tools` package, which is in the process of being deprecated.

See this article https://lwn.net/Articles/710533/

the `iproute2` tools, which includes `ip`, is the way of the world these days.

Reply:
Got it. Yes, then I'd agree with this. However "ip" is a generic binary name.. I've suggested above to verify that the ip tool is actually the one from the iproute2 package, and if not, then we fail back to ifconfig.

Reply:
I disagree that `ip` is a generic binary name. It's a standard linux thing. Any distro which has a different, non-compatible, command in its place is broken and should fix their setup.

Reply:
Linux isn't the only operating system out there (we verify binaries in other parts for darwin compatibility), and you also have to contend with the fact that these are local developer environments. I personally have a utility already on my darwin system that is named "ip" that is not the iproute2 utility.

I believe it's worth verifying that the "ip" binary you are going to call is the intended binary, and I've suggested a method of verifying the binary here: https://github.com/kubernetes/kubernetes/pull/53500#discussion_r143051468

Reply:
I don't think it's a good idea to try and second guess the user like that.
I would be okay with hiding it in an `if kube::build::is_osx` instead, though I could go either way since I know there are at least a couple OSX utilities which provide a sufficiently compatible `ip` command for `ip a`, which muddies the waters some.

I think you might be better to have your `ip` command be an alias that exists in your interactive shell, or if it was provided by a third party to let them know it's a bad name to use considering other shell scripts will make this same check.